back/engines/commercial/idea_assignment/spec/services/side_fx_idea_spec.rb

```diff
@@ -7,18 +7,29 @@
   let(:user) { create(:user) }
 
   describe 'before create' do
-    it 'sets the assignee to the default_assignee of the project' do
-      default_assignee = create(:admin)
-      project = create(:project, default_assignee: default_assignee)
-      idea = build(:idea, project: project)
+    let(:default_assignee) { create(:admin) }
+    let(:project) { create(:project, default_assignee: default_assignee) }
+
+    it 'does not set the assignee of a draft idea to the default_assignee of the project' do
+      idea = build(:idea, project: project, publication_status: 'draft')
+      service.before_create(idea, user)
+      expect(idea.assignee).to be_nil
+    end
+
+    it 'sets the assignee of a submitted idea to the default_assignee of the project' do
+      idea = build(:idea, project: project, publication_status: 'submitted')
+      service.before_create(idea, user)
+      expect(idea.assignee).to eq default_assignee
+    end
+
+    it 'sets the assignee of a published idea to the default_assignee of the project' do
+      idea = build(:idea, project: project, publication_status: 'published')
       service.before_create(idea, user)
       expect(idea.assignee).to eq default_assignee
     end
 
     it "doesn't change the assignee if it's already set" do
-      default_assignee = create(:admin)
       assignee = build(:admin)
-      project = create(:project, default_assignee: default_assignee)
```

Comment:
Why were these lines deleted? I think we need a project with a default assignee to make sure that this doesn't become the new assignee (if there's no default assignee it's less likely that the assignee would get changed).

Reply:
Sorry, I didn't see 😅